### ISSUE TITLE
[Enhancement] k8s-namespace-healthcheck needs GMP support

### ISSUE THREAD

**Issue by stewartshea** [runwhen-contrib/rw-public-codecollection#53]
**Observation** 
The k8s-namespace-healthcheck SLI currently combines some health checks from both the kube api and prometheus. As a user of GMP, we would need to support a google IAM service account similar to how it's used in the gcp-opsuite-promql codebundle. 

**Current Outcome** 
Currently this only supports direct prometheus with a header secret, which wouldn't work for GMP. 

**Desired Outcome**
My thought is a dropdown of the prometheus type, which would then help determine which secret is appropriate to use.

**Comment by stewartshea** [comment]
After reading through the code and some of the available prometheus health status that can be surfaced, I think we should split the k8s-namespace-healthcheck codebundle into 2 separate codebundles; 
1. Focused only on users with kubectl access (e.g. no prom instances)
2. Focused on prometheus metrics only

The thinking here is that requiring both integrations in the same codebundle excludes some use cases and makes additional use cases more complex (e.g. with multiple variations of managed prom instances (sysidg, google, grafana cloud, etc), and the kubectl codebundle can easily be refactored to use `kubectl get` instead of prometheus. 

@jon-funk what do you think?

**Comment by jon-funk** [comment]
I agree, splitting them up leads to more reusable codebundles and plays into having composite SLIs. This was a brief experiment in joining 2 functionalities together and while interesting - you can take it further with strategy patterns - it gets unwieldy fast and is antithetic to having simple code/codebundles.
